picard/acoustid/manager.py

```diff
@@ -106,37 +134,54 @@ def _batch_submit(self, submissions, errors=None):
                 log_msg, echo=None, timeout=3000)
             self._check_unsubmitted()
             return
-        submission_batch = submissions[:self.BATCH_SUBMIT_COUNT]
-        submissions = submissions[self.BATCH_SUBMIT_COUNT:]
-        fingerprints = [fingerprint for file_, fingerprint in submission_batch]
+
+        batch = list(self._batch(submissions))
+        if not batch:
+            if self.max_batch_size == 0:
+                log_msg = N_("AcoustID submission failed permanently, max batch size reduced to zero")
+            else:
+                log_msg = N_("AcoustID submission failed permanently, prolly too many retries")
+            log.error(log_msg)
+            self.tagger.window.set_statusbar_message(
+                log_msg, echo=None, timeout=3000)
+            self._check_unsubmitted()
+            return
+
         log.debug("AcoustID: submitting batch of %d fingerprints (%d remaining)...",
-            len(submission_batch), len(submissions))
+            len(batch), len(submissions))
         self.tagger.window.set_statusbar_message(
             N_('Submitting AcoustIDs ...'),
             echo=None
         )
         if not errors:
             errors = []
         next_func = partial(self._batch_submit, submissions)
-        self._acoustid_api.submit_acoustid_fingerprints(fingerprints,
-            partial(self._batch_submit_finished, submission_batch, errors, next_func))
+        self._acoustid_api.submit_acoustid_fingerprints(
+            [submission for file_, submission in batch],
+            partial(self._batch_submit_finished, batch, errors, next_func)
+        )
 
     def _batch_submit_finished(self, submissions, previous_errors, next_func, document, http, error):
         if error:
-            try:
-                error = load_json(document)
-                message = error["error"]["message"]
-            except BaseException:
-                message = ""
-            mparms = {
-                'error': http.errorString(),
-                'message': message
-            }
-            previous_errors.append(mparms)
-            log_msg = N_("AcoustID submission failed with error '%(error)s': %(message)s")
-            log.error(log_msg, mparms)
-            self.tagger.window.set_statusbar_message(
-                log_msg, mparms, echo=None, timeout=3000)
+            response_code = self._acoustid_api.webservice.http_response_code(http)
+            if response_code == 413:
+                self.max_batch_size = int(self.max_batch_size * self.BATCH_SIZE_REDUCTION_FACTOR)
+                log.warn("AcoustID: payload too large, batch size reduced to %d", self.max_batch_size)
```

Comment:
I guess here goes a call to something like 

```
        self._acoustid_api.submit_acoustid_fingerprints(
            [submission for file_, submission in submissions],
            partial(self._batch_submit_finished, submissions, previous_errors, next_func)
        )
```

Reply:
Not needed, see last iteration